src/main/java/ch/njol/skript/lang/function/Functions.java

```diff
@@ -227,51 +227,96 @@ private static Signature<?> signError(String error) {
+	 */
+	@Nullable
+	public static Function<?> getGlobalFunction(String name) {
+		Namespace namespace = globalFunctions.get(name);
+		if (namespace == null)
+			return null;
+		return namespace.getFunction(name, false);
+	}
+
+	/**
+	 * Gets a function, if it exists. Note that even if function exists in scripts,
+	 * it might not have been parsed yet. If you want to check for existence,
+	 * then use {@link #getLocalSignature(String, String)}.
 	 *
 	 * @param name Name of function.
 	 * @param script The script where the function is declared in. Used to get local functions.
 	 * @return Function, or null if it does not exist.
 	 */
 	@Nullable
-	public static Function<?> getFunction(String name, @Nullable String script) {
+	public static Function<?> getLocalFunction(String name, String script) {
 		Namespace namespace = null;
 		Function<?> function = null;
-		if (script != null) {
-			namespace = getScriptNamespace(script);
-			if (namespace != null)
-				function = namespace.getFunction(name);
-		}
-		if (namespace == null || function == null) {
-			namespace = globalFunctions.get(name);
-			if (namespace == null)
-				return null;
-			function = namespace.getFunction(name, false);
-		}
+		assert script != null;
+		namespace = getScriptNamespace(script);
+		if (namespace != null)
+			function = namespace.getFunction(name);
 		return function;
 	}
 
 	/**
 	 * Gets a signature of function with given name.
+	 * 
+	 * @deprecated in favour of {@link #getGlobalSignature(String)} for proper name.
+	 * @param name Name of function.
+	 * @return Signature, or null if function does not exist.
+	 */
+	@Deprecated
+	@Nullable
+	public static Signature<?> getSignature(String name) {
+		return getGlobalSignature(name);
+	}
+
+	/**
+	 * Gets a signature of function with given name.
+	 * 
+	 * @param name Name of function.
+	 * @return Signature, or null if function does not exist.
+	 */
+	@Nullable
+	public static Signature<?> getGlobalSignature(String name) {
+		Namespace namespace = globalFunctions.get(name);
+		if (namespace == null)
+			return null;
+		return namespace.getSignature(name, false);
+	}
+
+	/**
+	 * Gets a signature of function with given name.
+	 * 
 	 * @param name Name of function.
 	 * @param script The script where the function is declared in. Used to get local functions.
 	 * @return Signature, or null if function does not exist.
 	 */
 	@Nullable
-	public static Signature<?> getSignature(String name, @Nullable String script) {
+	public static Signature<?> getLocalSignature(String name, String script) {
 		Namespace namespace = null;
 		Signature<?> signature = null;
-		if (script != null) {
-			namespace = getScriptNamespace(script);
-			if (namespace != null)
-				signature = namespace.getSignature(name);
-		}
-		if (namespace == null || signature == null) {
-			namespace = globalFunctions.get(name);
-			if (namespace == null)
-				return null;
-			signature = namespace.getSignature(name, false);
-		}
+		assert script != null;
```

Comment:
I don't think the assertion is necessarily needed.

Reply:
Scrubbing the data, ensuring it's not null. It's always good to check the data for safety.

Reply:
Definitely - in this case though I think we are safe (especially since we don't check it anywhere else :))